dcap/ql/src/quote_generator.rs

```diff
@@ -6,18 +6,51 @@
 //! functionality will return errors.
 
 use mc_sgx_core_sys_types::sgx_target_info_t;
-use mc_sgx_core_types::TargetInfo;
+use mc_sgx_core_types::{Quote, Report, ReportData, TargetInfo};
 use mc_sgx_dcap_types::Quote3Error;
 use mc_sgx_util::ResultInto;
 
-/// Provides behavior to generate quotes for enclaves
+/// Functionality to create quotes.  Implementers should only need to implement
+/// `report` as that will need to come from the application enclave.
 pub trait QuoteGenerator {
+    /// Report from an application enclave
+    ///
+    /// # Arguments
+    /// * `target_info` - The information for the target enclave which will
+    ///   cryptographically verify the report. This is usually from the
+    ///   QE(Quoting Enclave).
+    /// * `report_data` - Report data used to communicate between enclaves
+    fn report(
```

Comment:
Curious about thoughts here? It felt like the simplest way to break the psuedo circular dependency since we need the target_info from the QE, to create a report, but then we need the report for the PCE to generate a quote.   
I'm not sure we should expose the QE vs PCE interfaces to the users as separate structs/traits

Reply:
I think this interface should be inverted to traits monkey-patched onto the structs in question. Like, `TargetInfo::for_quoting_enclave()`

Reply:
code changed, comment no longer applies